audio/sonic/files/patch-sonic-mac.diff

```diff
@@ -0,0 +1,119 @@
+--- Makefile.orig	2020-06-30 13:32:50.000000000 +0700
++++ Makefile	2020-06-30 14:09:51.000000000 +0700
+@@ -5,22 +5,42 @@
+ # safe.  We call malloc, and older Linux versions only linked in the thread-safe
+ # malloc if -pthread is specified.
+ 
+-SONAME=soname
++# Set this to 0 if you do not want to link in spectrogram generation.
++USE_SPECTROGRAM=1
++
++PREFIX=/usr/local
++BINDIR=$(PREFIX)/bin
++LIBDIR=$(PREFIX)/lib
++INCDIR=$(PREFIX)/include
++
++SONAME=soname,
++SHARED_OPT=-shared
++LIB_NAME=$(LIB_NAME)
++LIB_TAG=.0.3.0
+ UNAME := $(shell uname)
+ ifeq ($(UNAME), Darwin)
+-  SONAME=install_name
++  SONAME=install_name,$(LIBDIR)/
++  SHARED_OPT=-dynamiclib
++  LIB_NAME=libsonic.dylib
++  LIB_TAG=
+ endif
+-#CFLAGS=-Wall -g -ansi -fPIC -pthread
+-CFLAGS=-Wall -O3 -ansi -fPIC -pthread
+-LIB_TAG=0.2.0
+-CC=gcc
+-PREFIX=/usr
+-LIBDIR=$(PREFIX)/lib
++#CFLAGS=-Wall -Wno-unused-function -g -ansi -fPIC -pthread
++CFLAGS=-Wall -Wno-unused-function -O3 -ansi -fPIC -pthread
++SRC=sonic.c
++FFTLIB=
++ifeq ($(USE_SPECTROGRAM), 1)
++  CFLAGS+= -DSONIC_SPECTROGRAM
++  SRC+= spectrogram.c
++  FFTLIB=-lfftw3
++endif
++OBJ=$(SRC:.c=.o)
++
++# Set this to empty if not using spectrograms.
+ 
+-all: sonic libsonic.so.$(LIB_TAG) libsonic.a
++all: sonic $(LIB_NAME)$(LIB_TAG) libsonic.a
+ 
+-sonic: wave.o main.o libsonic.so.$(LIB_TAG)
+-	$(CC) $(CFLAGS) -o sonic wave.o main.o libsonic.so.$(LIB_TAG)
++sonic: wave.o main.o libsonic.a
++	$(CC) $(CFLAGS) -o sonic wave.o main.o libsonic.a -lm $(FFTLIB)
```

Comment:
You've changed sonic from using the shared library to using the static library. This means a copy of the static library will be embedded in the executable, which wastes disk space and is generally undesirable. This should continue to use the shared (dynamic on macOS) library.

This will require the LD_LIBRARY_PATH (DYLD_LIBRARY_PATH on macOS) environment variable to be set properly in the check target.

Reply:
That was from git head from the developer. I will check if the dylib can still be used here.

Reply:
Ok, it builds and tests with the dylib here.